### PR TITLE
Support multiple separators in StringTokenizer.

### PR DESCRIPTION
**Pull request by cesarbs**
cc @rynowak

### CODE REVIEW

File: src/Microsoft.Extensions.Primitives/StringTokenizer.cs [hunk 1]
```diff
@@ -66,14 +81,8 @@ public bool MoveNext()
                     Current = default(StringSegment);
                     return false;
                 }
-
-                var next = _value.IndexOf(_separator, _index);
```
Comment (rynowak) [inline src/Microsoft.Extensions.Primitives/StringTokenizer.cs]: This already exists: https://msdn.microsoft.com/en-us/library/6wh7x3fs(v=vs.110).aspx

**Comment by rynowak** [comment]
:shipit: after changing to built-in `IndexOfAny`

File: src/Microsoft.Extensions.Primitives/StringTokenizer.cs [hunk 2]
```diff
@@ -21,14 +21,29 @@ public struct StringTokenizer :  IEnumerable<StringSegment>
         /// <param name="value">The <c>string</c> to tokenize.</param>
         /// <param name="separator">The character to tokenize by.</param>
         public StringTokenizer(string value, char separator)
```
Comment (pranavkm) [inline src/Microsoft.Extensions.Primitives/StringTokenizer.cs]: Could we remove this? It's an extra array allocation that could be cached by the caller instead.

Comment (pranavkm) [inline src/Microsoft.Extensions.Primitives/StringTokenizer.cs]: Alternatively switch between single or array separator.

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**